README.md

```diff
@@ -1,102 +1,262 @@
 
-  app.get "/login", (request, response) ->
-    sp.create_login_url idp, (err, login_url) ->
-      return response.send 500, err if err?
-      response.location login_url
-      response.send 302, "Redirecting..."
+#### Example
+```javascript
 
-  app.get "/logout", (request, response) ->
-    sp.create_logout_url idp, (err, login_url) ->
-      return response.send 500, err if err?
-      response.location login_url
-      response.send 302, "Redirecting..."
+  // Initialize options object
+  var idp_options = {
+    sso_login_url: "https://idp.example.com/login",
+    sso_logout_url: "https://idp.example.com/logout",
+    certificates: [fs.readFileSync("cert-file1.crt").toString(), fs.readFileSync("cert-file2.crt").toString()],
+    force_authn: true,
+    sign_get_request: false,
+    allow_unencrypted_assertion: false
+  };
+
+  // Call identity provider constructor with options
+  var idp = saml2.IdentityProvider(idp_options);
+
+  // Example usage of identity provider.
+  // Pass identity provider into a service provider function with options and a callback.
+  sp.post_assert(idp, {}, callback);
 
-  app.post "/assert", (request, response) ->
-    sp.assert idp, response.body, (err, user) ->
-      response.send 500, err if err?
-      response.send 200, "Hello #{user.email}!"
 ```
 
 
-## Related Libraries
+## Example: Express implementation
+
+Library users will need to implement a set of URL endpoints, here is an example of [express](http://expressjs.com/) endpoints.
+
+```javascript
+
+  var express = require('express');
+  var app = express();
+
+  // Create service provider
+  var sp_options = {
+    entity_id: "https://sp.example.com/metadata.xml",
+    private_key: fs.readFileSync("key-file.pem").toString(),
+    certificate: fs.readFileSync("cert-file.crt").toString(),
+    assert_endpoint: "https://sp.example.com/assert"
+  };
+  var sp = saml2.ServiceProvider(sp_options);
+
+  // Create identity provider
+  var idp_options = {
+    sso_login_url: "https://idp.example.com/login",
+    sso_logout_url: "https://idp.example.com/logout",
+    certificates: [fs.readFileSync("cert-file1.crt").toString(), fs.readFileSync("cert-file2.crt").toString()]
+  };
+  var idp = saml2.IdentityProvider(idp_options);
+
+  // ------ Define express endpoints ------
+
+  // Endpoint to retrieve metadata
+  app.get("/metadata.xml", function(req, res) {
+    res.send(200, sp.create_metadata());
+  });
 
-- https://github.com/siphon-io/node-saml2
-- https://github.com/bozzltron/express-saml
-- https://github.com/bergie/passport-saml
-- https://github.com/auth0/node-saml
-- https://github.com/auth0/passport-wsfed-saml2
+  // Starting point for login
+  app.get("/login", function(req, res) {
+    sp.create_login_request_url(idp, {}, function(err, login_url, request_id) {
+      if (err != null)
+        return res.send(500);
+      // Save request_id in the session for future checking
+      req.session.request_id = request_id;
```

Comment:
We aren't using the session middleware so `req.session` isn't a thing. We should make note of this, or fix it.